### DISCUSSION TITLE
How to set a specific User View Page in the index page of resource

### DISCUSSION THREAD

**Discussion by SubhamKumar101201** [filamentphp/filament#10072] [Help] [answered]
i have a viewuser  page and i want to show a perticular user view page in the resource index page so, i try in the getPages() but i dont get the result by passing a record in the route it shows only one parameter allowed and if i try to set the record in the view page using mount function it give record needed in the route ? any idea how to get this

**Comment by leandrocfe** [comment]
try to add this method in the ViewPage

```php

public function mount(int | string $record): void
{
    $this->record = $this->resolveRecord(your_id);

    $this->authorizeAccess();

    if (! $this->hasInfolist()) {
        $this->fillForm();
    }
}

**Comment by SubhamKumar101201** [comment]
as per your suggestion I use this function but how to set the getPages in the resource 
'''      public static function getPages(): array
    {
        return [
            'index' => Pages\ViewStudent::route('/{record}/view'),
            'edit' => Pages\EditStudentDetail::route('/{record}/clinic-detail/edit')
        ];
    }      ''' but it gives me this errorMissing required parameter for [Route: filament.student.resources.settings.index] [URI: student/settings/{record}/view] [Missing parameter: record].

**Reply by leandrocfe** [reply]
You should use a view page
https://filamentphp.com/docs/3.x/panels/resources/viewing-records#creating-a-resource-with-a-view-page

**Reply by SubhamKumar101201** [reply]
??

**Comment by SubhamKumar101201** [comment]
Bro but if I don't add the index page it gives me error no index page present

**Comment by SubhamKumar101201** [comment]
I want when I click on navbar to the student resource it first open the view page then edit page no list and create page so i try to set the view page in index otherwise it doesn't work

**Comment by leoblanski** [comment]
Im with same problem here .... Didn't find a solution.

I have a resource that I want users to only edit. It doesn't make sense to show an index page with a list of resources. Therefore, I would like the "Tenant" menu, for example, to directly open the edit page with the form to make changes.

 I could not set the Pages with the ID because I have no information about session at this point.

Normal resource:
```
public static function getPages(): array
    {
        return [
            'index' => Pages\ListTeams::route('/'),
            'create' => Pages\CreateTeam::route('/create'),
            'edit' => Pages\EditTeam::route('/{record}/edit'),
        ];
    }
```

I want to do something like:

```
    public static function getPages(): array
    {
        return [
            'index' => Pages\ListTeams::route('/{record}/edit'),
        ];
    }
```

I've attempted to use custom pages that extend from Page and fill on mount function, but it returns errors.

`Missing required parameter for [Route: filament.user.resources.teams.index] [URI: teams/{record}/edit] [Missing parameter: record].`

It's because when Panel is mounted, doesn't have information about which ID Tenant will edit.

**Comment by leoblanski** [comment, marked as answer]
Heyyy, i found a solution.

It works as you are expecting.

```
    public static function getPages(): array
    {
        return [
            'edit' => Pages\EditTeam::route('/{record}/edit'),
            'index' => Pages\EditTeam::route("/settings/edit"),
        ];
    }
   ```
   
   ```
class EditTeam extends EditRecord
{
    protected static string $resource = TeamResource::class;

    public function mount(int | string $record): void
    {
        $this->record = auth()->user()->team;

        static::authorizeResourceAccess();
        $this->fillForm();
    }
    ....

```
